src/main/java/org/prebid/server/bidder/adhese/AdheseBidder.java

```diff
@@ -0,0 +1,279 @@
+         */
+        final List<BidderError> errors = new ArrayList<>();
+        final List<BidderBid> bidderBids = updateSeatBid.getBid().stream()
+                .filter(Objects::nonNull)
+                .map(bid -> makeBid(bid, bidRequest.getImp().get(0).getId(), errors))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+        return Result.of(bidderBids, errors);
+    }
+
+    private <T> List<T> decodeBodyToBidList(HttpCall<Void> httpCall, Class<T> bidClassName) {
+        try {
+            return mapper.mapper().readValue(
+                    httpCall.getResponse().getBody(),
+                    mapper.mapper().getTypeFactory().constructCollectionType(List.class, bidClassName));
+        } catch (DecodeException | JsonProcessingException e) {
+            throw new PreBidException(e.getMessage(), e);
+        }
+    }
+
+    private SeatBid convertAdheseBid(AdheseBid adheseBid, AdheseResponseExt adheseResponseExt,
+                                     AdheseOriginData adheseOriginData) {
+        final ObjectNode adheseExtJson = mapper.mapper().valueToTree(adheseOriginData);
+
+        return SeatBid.builder()
+                .bid(Collections.singletonList(Bid.builder()
+                        .id("1")
+                        .dealid(adheseResponseExt.getOrderId())
+                        .crid(adheseResponseExt.getId())
+                        .adm(getAdMarkup(adheseBid, adheseResponseExt))
+                        .ext(adheseExtJson)
+                        .build()))
+                .seat("")
+                .build();
+    }
+
+    private String getAdMarkup(AdheseBid adheseBid, AdheseResponseExt adheseResponseExt) {
+        if (Objects.equals(adheseResponseExt.getExt(), "js")) {
+            if (StringUtils.containsAny(adheseBid.getBody(), "<script", "<div", "<html")) {
+                String counter = "";
+                if (adheseResponseExt.getImpressionCounter().length() > 0) {
+                    counter = String.format("%s%s%s", "<img src='", adheseResponseExt.getImpressionCounter(),
+                            "' style='height:1px; width:1px; margin: -1px -1px; display:none;'/>");
+                }
+                return String.format("%s%s", adheseBid.getBody(), counter);
+            }
+            if (StringUtils.containsAny(adheseBid.getBody(), "<?xml", "<vast")) {
+                return adheseBid.getBody();
+            }
+        }
+        return adheseResponseExt.getTag();
+    }
+
+    private SeatBid convertAdheseOpenRtbBid(AdheseBid adheseBid) {
+        return (CollectionUtils.isNotEmpty(adheseBid.getOriginData().getSeatbid())
+                && CollectionUtils.isNotEmpty(adheseBid.getOriginData().getSeatbid().get(0).getBid()))
+                ? SeatBid.builder()
+                .bid(Collections.singletonList(Bid.builder().adm(adheseBid.getBody()).build()))
+                .build()
+                : null;
+    }
```

Comment:
You do not need `SeatBid` at all. Just return created Bid and check for it existence in the code instead of seatBid